contract/RadicleCloudBase.sol

```diff
@@ -0,0 +1,191 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity >=0.7.0 <0.9.0;
+
+/// @title RadicleCloudBase does on-chain accounting and emits events for operator.
+/// @dev RadicleCloudBase is an abstract contract.
+abstract contract RadicleCloudBase {
+    /// @dev Owner is needed to potentially re-adjust prices in future and withdraw funds.
+    address private owner;
+    /// @notice Keep track of expiry blocks for Orgs
+    mapping(address => uint64) public expiry;
+    /// @notice Remember owner of Org, one who initially created it
+    mapping(address => address) public orgOwner;
+
+    /// @notice Package price ber blockPerMonth
+    uint256 public pricePerMonth;
+    /// @notice Duration of blocks roughly equivalent to one month
+    uint32  public blocksPerMonth;
+
+    /// @notice MAXBLOCK is the maximum value of uint64
+    uint64 private constant MAXBLOCK = type(uint64).max;
+
+    /// @notice Array of when buys happened
+    uint64[] private buys    = [MAXBLOCK];
+    /// @notice Array of when expires happen
+    uint64[] private expires = [MAXBLOCK];
+    /// @notice Array of when cancels happened
+    uint64[] private cancels = [MAXBLOCK];
+    /// @notice Array of when cancel offsets happen
+    uint64[] private offsets = [MAXBLOCK];
+
+    /// @notice Tracks position of last impacted idx for all lasts
+    uint64[] private LAST = [0, 0, 0, 0, 0];
```

Comment:
Kind of confused by this. What is a "last"?

Also, convention is to only use CAPS for constants.